### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by fievelk**
Please review this PR so that I can eventually merge it.

This PR currently solves several test failures. Following are the main changes:
- merge PR by @ExplodingCabbage to fix `six` imports (#1467)
- refactor `six` imports so that they are correctly grouped following NLTK guidelines (see: [Package Structure](https://github.com/nltk/nltk/wiki/Package-Structure))
- add `pyparsing` and `python-crfsuite` packages to `tox` dependencies.
  
  This fixes the following test failures: 
  
  ```
  NameError: name 'pyparsing' is not defined
  NameError: name 'pycrfsuite' is not defined
  ```
- make `tox` aware of environment variables.
  
  This is needed, for example, to run tests in a system where custom environment variables (like `NLTK_DATA` or `SENNA`) have been set. See [Installing NLTK data](http://www.nltk.org/data.html) and [Installing Third Party Software](https://github.com/nltk/nltk/wiki/Installing-Third-Party-Software) for more info.
- replace Senna doctests with unit tests.
  
  Doctests were failing on systems that did not have a `Senna` executable in `/usr/share/senna-v3.0`. New tests check if a different location is set using the `SENNA` environment variable. If the executable folder cannot be found we expect that a `OSError` exception is raised. If this happens, tests are considered successful. **NOTE:** I suppose this behaviour can be debatable.
- Fix `aline.py` tests (see #1418)
  
  Doctests were failing with `python2.7` because of encoding problems. We now use unittest and add `from __future__ import unicode_literals` in the module. Thanks @geoffbacon for your help on the module.

Related issues:
- Better document how to run the tests (#1466) 
# NOTE

Some tests are still failing. Please submit other PR in order to fix them.
- Fix Gluesemantic tests (see #1050)
- Fix `ipipan` and `nkjp` corpora (see #1484)

@stevenbird @kmike @dimazest I would be grateful if you could please give me your opinion about this PR :)

### CODE REVIEW

File: nltk/test/unit/test_senna.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            result_2 = nertagger.tag('UN headquarters are in NY , USA .'.split())
+            expected_2 = [('UN', 'B-ORG'), ('headquarters', 'O'), ('are', 'O'),
+            ('in', 'O'), ('NY', 'B-LOC'), (',', 'O'), ('USA', 'B-LOC'), ('.', 'O')]
+            self.assertEqual(result_1, expected_1)
+            self.assertEqual(result_2, expected_2)
+        else:
+            print("OKKK")
+            print("OKKK")
+            print("OKKK")
+            print("OKKK")
```
Comment (kmike) [inline nltk/test/unit/test_senna.py]: what are these prints for?

Comment (fievelk) [reply nltk/test/unit/test_senna.py]: > what are these prints for?

Sorry, they were just for debugging :) I'm removing them.

**Comment by ExplodingCabbage** [comment]
> If the executable folder cannot be found we expect that a OSError exception is raised. If this happens, tests are considered successful. NOTE: I suppose this behaviour can be debatable.

These certainly seems less than ideal. I don't know anything about Senna and don't know much about Python's testing tools, but is there no way to either:
- Have running `tox` cause Senna to be installed automatically, or
- Have Senna tests be _skipped_ if Senna isn't installed, rather than passing

?

**Comment by fievelk** [comment]
@ExplodingCabbage Thank you. I agree with you especially on your second point. 

I'm not sure whether we can have all developers to have Senna and all other third party software installed in order to submit their code. But I think that skipping the test would be better than letting it successfully pass.

Please see my last commit a3725fa.

**Comment by stevenbird** [comment]
@fievelk – thanks for all your work here! Looks like a good solution for Senna; change to `aline.py` is just formatting, not refactoring, right?

**Comment by fievelk** [comment]
@stevenbird Yes, the first commit about `aline.py` is just formatting, while the second one contains fixes. I'll amend the commit message.
I'm also about to add some modifications made by @geoffbacon to `aline.py`. I'll push them soon.

**Comment by fievelk** [comment]
@stevenbird I think this PR is ready to be merged.
Please note that Gluesemantic tests are still occasionally failing because of non-deterministic output (see #1050).

**Comment by iliakur** [comment]
Sorry to jump in so late. Any reason why we aren't using the external `six` library?

**Comment by kmike** [comment]
@Copper-Head it seems @stevenbird bundled it in https://github.com/nltk/nltk/commit/f4342ffe451c2fa6e3f48f9646c5f2a92018a260; maybe install_requires didn't work for some users; I don't know the details.

**Comment by iliakur** [comment]
I found that commit as well, but couldn't find any discussion about it online and the commit message didn't mention _why_ this was necessary. Considering that `nltk.six` appears to be a copy/paste of the external library, I'm curious to know what could have prompted such a violation of DRY :)

**Comment by iliakur** [comment]
Ok, in the release notes for `3.1` is say this was done to 

> streamline installation on windows

Oh windows, the gift that keeps on giving...

**Comment by stevenbird** [comment]
@Cooper-Head I've removed nltk.six and added a dependency on six itself (https://github.com/nltk/nltk/commit/90ac1bf13f8ecd7842d9cc39970631dcbba97ea8). Let's hope there's no problems for Windows installation any more.

**Comment by stevenbird** [comment]
I'm finding it quite difficult to review this PR as so many unrelated things are going on in it.

**Comment by fievelk** [comment]
@stevenbird If you find it easier to review, I'll split it in multiple PR then as soon as I have time.

**Comment by stevenbird** [comment]
@fievelk – yes that would be super helpful!

**Comment by fievelk** [comment]
@stevenbird I split this PR into several smaller PRs, hoping this will help you to review them: 
- #1497 
- #1499
- #1500
- #1501

I think this PR can be closed.

**Comment by stevenbird** [comment]
Thanks @fievelk!

**Pull request closed without merging**